quinn-proto/src/coding.rs

```diff
@@ -1,18 +1,25 @@
+//! Coding related traits.
+
 use std::net::{Ipv4Addr, Ipv6Addr};
 
 use bytes::{Buf, BufMut};
 use thiserror::Error;
 
 use crate::VarInt;
 
+/// Unexpected end of buffer error.
 #[derive(Error, Debug, Copy, Clone, Eq, PartialEq)]
 #[error("unexpected end of buffer")]
 pub struct UnexpectedEnd;
 
+#[doc(hidden)]
 pub type Result<T> = ::std::result::Result<T, UnexpectedEnd>;
 
+/// Trait to handle encoding and decoding from byte buffers.
 pub trait Codec: Sized {
+    /// Try to decode `Self` from the provided buffer.
     fn decode<B: Buf>(buf: &mut B) -> Result<Self>;
+    /// Encode `self` into the provided buffer.
```

Comment:
We should document exactly where the bytes end up in the buffer. A user might reasonably assume that they get written at the start.
```suggestion
    /// Append the encoding of `self` to the provided buffer
```